### PR TITLE
Golang assignment 7

### PR DESCRIPTION
**Pull request by divyatambat**
1. Given a string containing conversation between alice and bob. In the string, if it reaches $, it means end of alice message and if it reaches #, it means end of bob's message. and if it reaches ^,
it means end of conversation ignore string after that.

Note: given string doesn't contain any spaces. If message contains two continuous conversations from one person it should be printed one after another as given in the example. 

write a program to separate out messages from alice and bob. Write messages from alice and bob on seperate channels, whenever a message from alice/bob, print it in front of their name as shown in the example below:

Note: there is a single space before and after colon(:) and no space before and after comma.

e.g: "helloBob$helloalice#howareyou?#Iamgood.howareyou?$^"
output: alice : helloBob,bob : helloalice,bob : howareyou?,alice : Iamgood.howareyou?

2. Given a string, reverse it using one go routine.And inside go routine print reversed string and number of goroutines launched

e.g: Input: test123 output: 321tset 2

### CODE REVIEW

File: Asg0.7-2.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		runes := []rune(originalString)
+
+		for currentIndex := 0; currentIndex < len(runes)/2; currentIndex++ {
+			oppositeIndex := len(runes) - 1 - currentIndex
```
Comment (yama11299) [inline Asg0.7-2.go]: Can you reduce the number of len() calls to optimise this code?

Comment (divyatambat) [reply Asg0.7-2.go]: okay, I'll try

Comment (yama11299) [reply Asg0.7-2.go]: Nooo.. This hasn't optimised but increased memory footprint.
try to remove len() call that is inside of for loop

Comment (yama11299) [reply Asg0.7-2.go]: Superb

**Pull request merged**